document_merge_service/api/authentication.py

```diff
@@ -0,0 +1,109 @@
+import requests
+from django.conf import settings
+from django.core.cache import cache
+from django.utils.encoding import smart_text
+from django.utils.translation import ugettext as _
+from jose import ExpiredSignatureError, JWTError, jwt
+from rest_framework import authentication, exceptions
+
+
+class AnonymousUser(object):
+    def __init__(self):
+        self.username = None
+        self.groups = []
+
+    @property
+    def group(self):
+        return self.groups and self.groups[0]
+
+    @property
+    def is_authenticated(self):
+        return False
+
+    def __str__(self):
+        return "AnonymousUser"
+
+
+class OIDCUser(AnonymousUser):
+    def __init__(self, decoded_token):
+        self.username = decoded_token["sub"]
+        self.groups = decoded_token.get(settings.OIDC_GROUPS_CLAIM) or []
```

Comment:
```suggestion
        self.groups = decoded_token.get(settings.OIDC_GROUPS_CLAIM, [])
```

Reply:
This is done by purpose because a OIDC claim is out of our control and they might simply set it to null which needs to be safe guarded to empty list within Caluma.